packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -137,25 +132,45 @@ class BottomNavigationBar extends StatefulWidget {
   /// If [type] is null then [BottomNavigationBarType.fixed] is used when there
   /// are two or three [items], [BottomNavigationBarType.shifting] otherwise.
   ///
-  /// If [fixedColor] is null then the theme's primary color,
-  /// [ThemeData.primaryColor], is used. However if [BottomNavigationBar.type] is
-  /// [BottomNavigationBarType.shifting] then [fixedColor] is ignored.
+  /// Prefer [selectedItemColor] over [fixedColor], since [selectedItemColor]
+  /// applies to both shifting and fixed selected item colors. If
+  /// [selectedItemColor] is null, [fixedColor] will be used for fixed
+  /// [BottomNavigationBar]s. If [selectedItemColor] and [fixedColor] are
+  /// null, then the theme's primary color, [ThemeData.primaryColor] is used.
+  ///
+  /// [iconSize] must be at least 8.0 for the [Icon]s to display properly.
+  ///
+  /// [selectedFontSize], [unselectedFontSize], and [elevation] must be non-null
+  /// and non-negative.
+  ///
+  /// [showUnselectedLabels] must not be null.
```

Comment:
I like the `must be non-null` you say above.

Reply:
Done.